src/agent/utilities.ts

```diff
@@ -109,16 +114,21 @@ export function readDirectory(directory: string, includeFiles: boolean, includeF
                     Q.nfcall(fs.stat, fullPath)
                         .then((stat: fs.Stats) => {
                             if (stat && stat.isDirectory()) {
-                                readDirectory(fullPath, includeFiles, includeFolders)
-                                    .then((moreFiles: string[]) => {
-                                        results = results.concat(moreFiles);
-                                        if (--count === 0) {
-                                            deferred.resolve(results);
-                                        }
-                                    },
-                                    (error) => {
-                                        deferred.reject(new Error(error.toString()));
-                                    });
+                                if (searchOption && (searchOption == SearchOption.TopDirectoryOnly)) {
```

Comment:
searchOption is declared as "nullable", i.e. searchOption?: SearchOption. I hoped that the condition sais: if ((searchOption != null) && (searchOption == SearchOption.TopDirectoryOnly)).
Am I wrong?